src/Agent.Listener/Agent.cs

```diff
@@ -191,5 +204,19 @@ private void PrintUsage()
         {
             _term.WriteLine(StringUtil.Loc("ListenerHelp"));
         }
+
+        private void Quit()
+        {
+            _term.WriteLine("Exiting...");
+            if (_inConfigStage)
+            {
+                HostContext.Dispose();
```

Comment:
I had another exit flow implemented, where Terminal class throws OperationCanceledException in response to CTRL-C, but it requires quite some code in to unblock Console.ReadKey. Calling HostContext.Dispose here seems simpler and works. I can see the agent log file is flushed (if I don't call Dispose is it empty)